iothub/device/src/Transport/Mqtt/OrderedTwoPhaseWorkQueue.cs

```diff
@@ -15,71 +16,79 @@ private class IncompleteWorkItem
         {
             public IncompleteWorkItem(TWorkId id, TWork workItem)
             {
-                this.WorkItem = workItem;
-                this.Id = id;
+                WorkItem = workItem ?? throw new ArgumentNullException(nameof(workItem));
+                Id = id ?? throw new ArgumentNullException(nameof(id));
             }
 
             public TWork WorkItem { get; }
 
             public TWorkId Id { get; }
         }
 
-        private readonly Func<TWork, TWorkId> getWorkId;
-        private readonly Func<IChannelHandlerContext, TWork, Task> completeWork;
-        private readonly Queue<IncompleteWorkItem> incompleteQueue = new Queue<IncompleteWorkItem>();
+        private readonly Func<TWork, TWorkId> _getWorkId;
+        private readonly Func<IChannelHandlerContext, TWork, Task> _completeWork;
+        private readonly Queue<IncompleteWorkItem> _incompleteQueue = new Queue<IncompleteWorkItem>();
 
-        public OrderedTwoPhaseWorkQueue(Func<IChannelHandlerContext, TWork, Task> worker, Func<TWork, TWorkId> getWorkId, Func<IChannelHandlerContext, TWork, Task> completeWork)
+        public OrderedTwoPhaseWorkQueue(
+            Func<IChannelHandlerContext, TWork, Task> worker,
+            Func<TWork, TWorkId> getWorkId,
+            Func<IChannelHandlerContext, TWork, Task> completeWork)
             : base(worker)
         {
-            this.getWorkId = getWorkId;
-            this.completeWork = completeWork;
+            _getWorkId = getWorkId ?? throw new ArgumentNullException(nameof(getWorkId));
+            _completeWork = completeWork ?? throw new ArgumentNullException(nameof(completeWork));
         }
 
         public Task CompleteWorkAsync(IChannelHandlerContext context, TWorkId workId)
         {
-            if (this.incompleteQueue.Count == 0)
+            if (!_incompleteQueue.Any())
             {
                 throw new IotHubException("Nothing to complete.", isTransient: false);
             }
-            IncompleteWorkItem incompleteWorkItem = this.incompleteQueue.Peek();
-            if (incompleteWorkItem.Id.Equals(workId))
+
+            IncompleteWorkItem incompleteWorkItem = _incompleteQueue.Peek();
+            if (incompleteWorkItem != null
```

Comment:
No guarantee that `Peek` returns an item, right?

Reply:
From the docs it will throw an exception if the queue is empty. But it will return what ever is there.

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.queue-1.peek?view=netcore-3.1